docs/spec/concepts.rst

```diff
@@ -18,17 +278,34 @@ Example::
           e = [e]
       ...
 
-A type factored by ``T1 | T2 | ...`` is a supertype
-of all types ``T1``, ``T2``, etc., so that a value that
-is a member of one of these types is acceptable for an argument
-annotated by ``T1 | T2 | ...``.
+A union type ``T1 | T2``, where ``T1`` and ``T2`` are fully static types,
+represents the set of values formed by the union of all the sets of values
+represented by ``T1`` and ``T2``. Thus, by the definition of the supertype
+relation, the union ``T1 | T2`` is a supertype of both ``T1`` and ``T2``. Via
+materialization, the gradual types ``S1`` and ``S2`` are both assignable to a
+gradual union type ``S1 | S2``.
 
-One common case of union types are *optional* types.  By default,
-``None`` is an invalid value for any type, unless a default value of
-``None`` has been provided in the function definition.  Examples::
+If ``B`` is a subtype of ``A``, ``B | A`` is equivalent to ``A``.
+
+This rule applies only to subtypes, not assignable-to. The union ``T | Any`` is
```

Comment:
I'm so glad that we're finally clarifying this point — and that the formulation is well justified by the theoretical framework.